### PR TITLE
dev: switch to npm workspaces

### PR DESCRIPTION
**Pull request by beejjacobs**
## Ticket Number

https://pins-ds.atlassian.net/browse/AAPD-809

## Description of change

Moving to npm workspaces:

- remove individual package-lock files
- update Dockerfiles, pipelines, and builds to support workspaces
- update docs
- some fixes to forms-web-app to use `require.resolve`
- added placeholder `web-comment` package (fixed issue with `webpack` being hoisted by `webpack-cli` not being hoisted

Important packages with changes (yes I reviewed all/most package-lock changes...):

- `@azure/storage-blob` - 12.16.0 (bump from 12.10.0 for docs-api)
- `express`  - 4.18.2 (bump from 4.18.1 for some packages)
- `mongodb` - api uses 3.7.4, forms uses 3.6.12, docs uses 3.7.3 - unchanged
- `notifications-node-client` - 5.2.3 (bump from 5.1.1)
- `nunjucks` - 3.2.4 (bump from 3.2.3)
- `webpack` - 5.89.0 (bump from 5.76.1)

Tested locally. Need to check functions build + deploy OK - script(s) for build copied from back-office.

## Checklist

- [ ] Requires infrastructure changes
- [ ] If adding or remove environment variables (e.g. in `docker-compose.yaml`) then I have updated the appropriate Helm chart
- [x] I have updated the documentation accordingly
- [x] My commit history in this PR is linear
- [ ] New features have tests
- [ ] Breaking change (team conversation required)

## Important

Please do not merge from `main` (please only [rebase](https://github.com/foundry4/appeal-planning-decision/wiki/An-intro-to-Git-Rebase)). This keeps the history linear and easier to debug.

### CODE REVIEW

**Comment by beejjacobs** [comment]
Need to fix tests - something not working with `jest-fetch-mock`.

**Review by chrishawldev — DISMISSED** [review]
Looks good to me. Would be interested to see if there is much difference in size in docker images, e.g is it more or less efficient or no real difference.

**Comment by beejjacobs** [comment]
> Looks good to me. Would be interested to see if there is much difference in size in docker images, e.g is it more or less efficient or no real difference.

Good question. Will be interesting to see. Here's what looks like a typical output currently: 
image

File: docker-compose.yml [hunk 1]
```diff
@@ -281,6 +289,8 @@ services:
     depends_on:
```
Comment (bryn-pins) [inline docker-compose.yml:289]: appeals-service-api-data:
    build:
      context: ./data
      
 Given we're in ./data
 I don't think the adding of package-lock works

Comment (beejjacobs) [reply docker-compose.yml:290]: Yes good spot, I had build from root not checked docker-compose. Sorted in next commit.

**Comment by beejjacobs** [comment]
Fixed `data` and `mocked-services` Docker builds now, I think!

**Comment by beejjacobs** [comment]
First successful build: 
image

Looks like forms-web-app is quite a bit large. Might be worth some time to check that.

**Comment by chrishawldev** [comment]
> First successful build: image
> 
> Looks like forms-web-app is quite a bit large. Might be worth some time to check that.

Is this because we're now copying in the package lock from the root workspace like
COPY package.json /opt/
COPY package-lock.json /opt/
Which in theory will contain dependencies for other modules as well?
I don't think this is that big of an issue, best practise says keep your containers as small as possible but in reality having an extra 150Mb on the container size to have a pretty quick win with using npm workspaces is a good trade off IMHO and it's something we can come back to with a separate task in the future.

**Review by chrishawldev — APPROVED** [review]

File: docker-compose.yml
Comment (bryn-pins) [inline docker-compose.yml:1]: Did we want to add the azurite version?
image: mcr.microsoft.com/azure-storage/azurite:3.26.0

**Review by bryn-pins — APPROVED** [review]

**Pull request merged**